kolibri/logger/migrations/0004_tidy_progress_range.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.10 on 2018-03-28 07:50
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+def tidy_progress_range(apps, schema_editor):
+    """
+    Tidies progress ranges because a bug had caused them to go out of range
+    """
+    ContentSessionLog = apps.get_model('logger', 'ContentSessionLog')
+    ContentSummaryLog = apps.get_model('logger', 'ContentSummaryLog')
+
+    # Not knowing how floating points will behave in the local database,
+    # 1.0 might become bigger than 1.0!!
+    ContentSessionLog.objects.filter(progress__lt=0).update(progress=0.0)
+    ContentSessionLog.objects.filter(progress__gt=1).update(progress=1.0)
+    ContentSummaryLog.objects.filter(progress__lt=0).update(progress=0.0)
+    ContentSummaryLog.objects.filter(progress__gt=1).update(progress=1.0)
```

Comment:
Does calling `update` from within a migration use the model's custom model manager? Need to confirm, otherwise this won't mark the models as "dirty" for Morango.

Reply:
Using `update` doesn't invoke signals or `save` methods on instances, no.. it's pure SQL so to say.

Reply:
Right, I know it doesn't do either of those. But we override `update` on Morango-synced model managers. I'm assuming this still goes through that:
https://github.com/learningequality/morango/blob/master/morango/query.py#L15

Reply:
>Set the dirty bit manually -- is it just hard-coded to _morango_dirty_bit

Yeah, that should work fine -- based on the filtering, we know the values will all be updated anyway, and hence will necessarily be dirty.